### PR TITLE
Feature/task 269 keycloak

### PR DESCRIPTION
**Pull request by MarcoGoC**
# Description 📝
Connect React front-end with keycloak service   Tasks 267 - 271
Login button on the home page along with a title change to display the name and log-off button 

## Taiga Link 🔗
https://taiga.dts-stn.com/project/ericdube-hfp-shared-backlog/task/267?kanban-status=131

## Checklist ✅
- [x] created story for all visual components
- [x] created or modified necessary unit tests
- [ ] created or modified e2e tests
- [ ] modified README or any applicable documentation on the changes you made
- [ ] mobile version tested

### CODE REVIEW

**Review by Moro-Code — CHANGES_REQUESTED** [review]
Hey marco, there should only be changes on the home page regarding adding any components needed to log in. Could you do a pr from main first into your branch. It makes it difficult to do code reviews when there any many unrelated changes

File: src/pages/home.js [hunk 1]
```diff
@@ -67,11 +77,7 @@ export function Home() {
 
   // effect to initially fetch count when component mounts
   useEffect(() => {
-    if (
-      !triedFetchedBenefitsCount &&
-      !isFetchingBenefitsCount &&
-      !fetchBenefitsCountFailed
-    ) {
+    if (!triedFetchedBenefitsCount && !isFetchingBenefitsCount) {
```
Comment (Moro-Code) [inline src/pages/home.js]: Why is this being modified. This is related to other tasks. Does this need to be modified for keycloak ?

File: src/pages/home.js [hunk 2]
```diff
@@ -108,17 +114,37 @@ export function Home() {
           fetchBenefitsFailed
             ? fetchBenefitsFailedObj
             : fetchBenefitsCountFailedObj
```
Comment (Moro-Code) [inline src/pages/home.js:120]: why is this being changed

File: src/variables.js [hunk 3]
```diff
@@ -4,8 +4,8 @@ export const ENVIRONMENT = process.env.NODE_ENV;
 let benefitservice_env_url = process.env.REACT_APP_BENEFITSERVICE_BASE_URL;
 
 if (!benefitservice_env_url && ENVIRONMENT !== "production") {
-  //  benefitservice_env_url = "http://localhost:1337";
-  // this uses the published version of strapi until we get the benefit service working.
+  benefitservice_env_url = "http://localhost:1337";
+  benefitservice_env_url = "https://benefit-service-dev.dev.dts-stn.com";
```
Comment (Moro-Code) [inline src/variables.js]: This is not related to keycloak. Why is this being changed in this PR

File: src/keycloak.js [hunk 4]
```diff
@@ -0,0 +1,12 @@
+import Keycloak from "keycloak-js";
+
+// Setup Keycloak instance as needed
+// Pass initialization options as required  TODO
+
+const keycloak = new Keycloak({
```
Comment (Moro-Code) [inline src/keycloak.js:7]: Please do not hardcode things. Better to add to the variables file and add defaults for development perhaps

File: src/components/molecules/TitleUserLogout.js [hunk 5]
```diff
@@ -0,0 +1,75 @@
...
+import PropTypes from "prop-types";
+import { ActionButton } from "../atoms/ActionButton";
+import { Title } from "../atoms/Title";
+
+/**
+ * This component is for the login button :)
+ */
+export function TitleUserLogout(props) {
+  const handleClick = (event) => props.onClick(event.currentTarget.id);
+  const userName = props.userName;
```
Comment (Moro-Code) [inline src/components/molecules/TitleUserLogout.js:11]: nitpick: unnecessary variables here

File: src/components/molecules/Login.js [hunk 6]
```diff
@@ -0,0 +1,38 @@
+import React from "react";
+import PropTypes from "prop-types";
+import { ActionButton } from "../atoms/ActionButton";
+
+/**
+ * This component is for the login button :)
+ */
+export function Login(props) {
```
Comment (Moro-Code) [inline src/components/molecules/Login.js:11]: this component is really not needed in my opinion just use ActionButton component in header component and it will be available everywhere

File: src/components/molecules/TitleUserLogout.js [hunk 7]
```diff
@@ -0,0 +1,75 @@
+import React from "react";
+import PropTypes from "prop-types";
+import { ActionButton } from "../atoms/ActionButton";
+import { Title } from "../atoms/Title";
+
+/**
+ * This component is for the login button :)
+ */
+export function TitleUserLogout(props) {
```
Comment (Moro-Code) [inline src/components/molecules/TitleUserLogout.js:9]: Logout button and welcome message should be added to header not title component ? 
Screen Shot 2021-01-18 at 11 42 45 AM

Should be here 
Screen Shot 2021-01-18 at 11 43 23 AM

File: src/App.js [hunk 8]
```diff
@@ -20,7 +23,11 @@ function App() {
     }
   }, [i18n, language]);
 
-  return <Routes />;
+  return (
+    <ReactKeycloakProvider authClient={keycloak}>
```
Comment (Moro-Code) [inline src/App.js]: Screen Shot 2021-01-18 at 11 45 55 AM

See documentation. Any other providers should be within this one. recommend adding it to the index.js instead

**Review by Moro-Code — DISMISSED** [review]

File: src/components/molecules/Login.js (hunk 6 above)
Comment (MarcoGoC) [reply src/components/molecules/Login.js:11]: agree is mostly for future use (next sprint) box and text from the design

File: src/components/molecules/TitleUserLogout.js (hunk 7 above)
Comment (MarcoGoC) [reply src/components/molecules/TitleUserLogout.js:9]: that's what I have been asking for the last 4 days. and this is where they said to add it

File: src/components/molecules/TitleUserLogout.js (hunk 5 above)
Comment (MarcoGoC) [reply src/components/molecules/TitleUserLogout.js:11]: agree

File: src/variables.js (hunk 3 above)
Comment (MarcoGoC) [reply src/variables.js]: this is what it is on main, mostly

File: src/keycloak.js (hunk 4 above)
Comment (MarcoGoC) [reply src/keycloak.js:7]: next pr

File: src/pages/home.js (hunk 1 above)
Comment (MarcoGoC) [reply src/pages/home.js:80]: good catch

File: src/pages/home.js (hunk 2 above)
Comment (MarcoGoC) [reply src/pages/home.js:120]: the lifejourneys were removed from main

File: src/pages/home.js (hunk 1 above)
Comment (MarcoGoC) [reply src/pages/home.js]: no,  lifejourney were removed from the main, and after some git problems I had to redo everything 
the "!fetchBenefitsCountFailed" was remove my mistake

File: src/App.js (hunk 8 above)
Comment (MarcoGoC) [reply src/App.js]: ok

File: src/pages/home.js (hunk 2 above)
Comment (Moro-Code) [reply src/pages/home.js:120]: As part of another task not this one. PR's should be limited according to what you are working on

File: src/variables.js (hunk 3 above)
Comment (Moro-Code) [reply src/variables.js]: regardless your branch is touching code that are not part of what you are implementing. If its too hard to revert recommend taking your components and staring again from main.

File: src/keycloak.js (hunk 4 above)
Comment (Moro-Code) [reply src/keycloak.js:7]: fix it in this one please as part of this review

File: src/components/molecules/TitleUserLogout.js (hunk 7 above)
Comment (Moro-Code) [reply src/components/molecules/TitleUserLogout.js:9]: I believe during when we talked about the story we said it would be in the header. In most applications this is where it should be. In any case I wasn't made aware that you did not know this so my apologies if things were not clear as I thought they should have been

Comment (MarcoGoC) [reply src/components/molecules/TitleUserLogout.js:9]: no need to apologize, I understand your point and I missed when the story was being discussed. thanks

**Review by will0684 — APPROVED** [review]
Looks good to me!
